Dockerfile is added to the codebase

ai-naymul opened the pull request:
The dockerfile is added which use the Gurobi docker image

Fixes #78

ai-naymul committed:
Dockerfile is added to the codebase

The dockerfile is added which use the Gurobi docker image

vissarion commented:
@ai-naymul thanks for this PR. 

I think it is better if we add that file inside `doc` maybe in a directory called `docker` or something that will contain that file and a README with instructions on how to use it. As for example, https://volesti.readthedocs.io/en/latest/getting_started/install.html#development-environment-from-docker-container

@hariszaf what do you think?

@ai-naymul ideally a github action could be added to test that the dockerfile works as expected.

ai-naymul commented:
> @ai-naymul thanks for this PR.
> 
> I think it is better if we add that file inside `doc` maybe in a directory called `docker` or something that will contain that file and a README with instructions on how to use it. As for example, https://volesti.readthedocs.io/en/latest/getting_started/install.html#development-environment-from-docker-container
> 
> @hariszaf what do you think?
> 
> @ai-naymul ideally a github action could be added to test that the dockerfile works as expected.

Yeah we should add that instructions stuff after the installation section in the README file...

ai-naymul commented:
@vissarion should I work on the readme file instruction part of these PR right now and make a commit in the same pr or make another PR regarding that?

vissarion commented:
I think it is better to do it in this PR. Do you confirm that the docker file you wrote works? What are the commands such that we can reproduce that? Are you able to build dingo in that container and run the tests?

hariszaf commented on dockerfile:
```diff
@@ -0,0 +1,20 @@
+    && rm -rf /var/lib/apt/lists/*
+
+# Install dingo
+RUN pip install dingo
```
Your Dockerfile fails in this line. 
This is because `dingo` is not on PyPi. 

So, maybe, you could follow the installation instructions? 
For example, 
- get dependencies 
```
apt-get install libsuitesparse-dev &> /dev/null
pip install sparseqr 
pip install Cython 
pip install cobra  
pip install kaleido
```
- clone `dingo` repo
```
git clone https://github.com/GeomScale/dingo.git
cd dingo/
git submodule update --init
```
- get `boost` library:
```
wget -O boost_1_76_0.tar.bz2 https://boostorg.jfrog.io/artifactory/main/release/1.76.0/source/boost_1_76_0.tar.bz2 
tar xjf boost_1_76_0.tar.bz2 &> /dev/null
rm boost_1_76_0.tar.bz2 &> /dev/null
```
- Install `dingo`
```
python setup.py install --user
```

I think this might work. 



 

first clone the `dingo` repo on the image, then run the `init submodule` command, get  and then follow

hariszaf reviewed (changes requested):
I tried this Dockerfile and its current version fails. 
Maybe if you follow the steps described inline, it might work.

hariszaf commented:
Hi @vissarion and @ai-naymul 

[here](https://github.com/GeomScale/dingo/commit/50f90afd545c51e1fc0a351bb1915758876ddc29) you may find a working Dockerfile. 

as I do not know if that's in a best-practices way, please have a look and share thoughts. :taco:

ai-naymul commented:
> I think it is better to do it in this PR. Do you confirm that the docker file you wrote works? What are the commands such that we can reproduce that? Are you able to build dingo in that container and run the tests?

Okay I will make the changes

ai-naymul commented:
> I tried this Dockerfile and its current version fails. Maybe if you follow the steps described inline, it might work.

Oh..I didn't notice that the gurobi is not in pypi library , I am sorry I will fix this asap

ai-naymul committed:
docker file is updated and instruction added into readme file

Due to unavailability the docker file is failed to run before now that sloved and instruction for using the dockerfile is added

ai-naymul commented:
@hariszaf @vissarion could you please check the recent commit...!!
